File: FruitBasket.java
public class FruitBasket {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // TODO code application logic here
     Scanner scr = new Scanner(System.in);
        Stack<String> baskets = new Stack<>();

        String A = "apple";
        String O = "orange";
        String M = "mango";
        String G = "guava";

        System.out.println("Catch and eat any of these fruits: ('apple', 'orange', 'mango', 'guava')");
        System.out.print("How many fruits would you like to catch?: ");
        int catcher = scr.nextInt();
        scr.nextLine(); // Consume the newline character

        for (int x = 1; x <= catcher; x++) {
            System.out.println("Fruit " + x + " of " + catcher + ":");
            char fruit = scr.next().charAt(0);

            switch (Character.toLowerCase(fruit)) {
                case 'a':
                    baskets.push(A);
                    break;
                case 'o':
                    baskets.push(O);
                    break;
                case 'm':
                    baskets.push(M);
                    break;
                case 'g':
                    baskets.push(G);
                    break;
                default:
                    System.out.println("Invalid fruit, try again.");
                    x--; // Decrement x to repeat the iteration
                    continue; // Skip to the next iteration
            }
            System.out.println("Your basket now has : " + baskets);
        }

        // Eating fruits
        while (!baskets.isEmpty()) {
            System.out.print("Press 'E' to eat a fruit ");
            String action = scr.nextLine();

            if (action.equalsIgnoreCase("E")) {
                String eatenFruit = baskets.pop();
                System.out.println("You are Eat a " + eatenFruit + "!");
                System.out.println("Fruits in the Basket: " + baskets);
            } else {
                System.out.println("or you don't want to eat");
            }
        }

        System.out.println("No more fruits.");
    }
}

File: fruit_basket.py
def main():
    baskets = []

    A = "apple"
    O = "orange"
    M = "mango"
    G = "guava"

    print("Catch and eat any of these fruits: ('apple', 'orange', 'mango', 'guava')")
    catcher = int(input("How many fruits would you like to catch?: "))

    for x in range(1, catcher + 1):
        fruit = input(f"Fruit {x} of {catcher}: ").strip().lower()

        if fruit == 'a':
            baskets.append(A)
        elif fruit == 'o':
            baskets.append(O)
        elif fruit == 'm':
            baskets.append(M)
        elif fruit == 'g':
            baskets.append(G)
        else:
            print("Invalid fruit, try again.")
            continue  # Skip to the next iteration

        print("Your basket now has:", baskets)

    # Eating fruits
    while baskets:
        action = input("Press 'E' to eat a fruit or  you don't want to eat ").strip().lower()

        if action == 'e':
            eaten_fruit = baskets.pop()
            print("You are Eat a ", eaten_fruit + "!")
            print("Fruits in the Basket:", baskets)
        else:
            print("You don't want to eat.")

    print("No more fruits.")

if __name__ == "__main__":
    main()
